upgrade.py

```diff
@@ -201,6 +208,8 @@ def __init__(self, source):
         self.safe2upgrade = os.path.isfile(safe2upgrade_path)
         self.vgs_output = None
 
+        self.safe2mbrupgrade = os.path.isfile(os.path.join(primary_fs.mount_point, constants.SAFE_2_MBR_UPGRADE))
```

Comment:
A feature flag would be appropriate if choosing to enable the functionality in the host installer. However, this is a bit different. The upgrade is enabled based on the presence of the marker file in the installation itself. This may or may not be present depending on the component. It is expected that the component that initiates the update sets the marker file if needed. This follows the established convention for partition layout upgrades.